### PR TITLE
Update legacy Repository usage in various example files

### PR DESCRIPTION
**Pull request by Hvanderwilk**
# What does this PR do?
The usage in the example is marked for deprecation here https://huggingface.co/docs/huggingface_hub/guides/upload#legacy-upload-files-with-git-lfs. Use the new recommended methods.

I've further removed the `epoch_/step_` gitignores, that seems a bit against the purpose of doing the intermediate push, for which some examples store models at an epoch.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by amyeroberts — COMMENTED** [review]
Thanks for updating! Could you extend this to all the other examples which use `Repository`?

**Comment by Hvanderwilk** [comment]
> Thanks for updating! Could you extend this to all the other examples which use `Repository`?

@amyeroberts sure! Could you point out if I can find why the CI actions are failing? In the details it just shows as failed without any details. I ran the quality tools / tests locally and all seemed ok.

File: examples/flax/image-captioning/run_image_captioning_flax.py [hunk 1]
```diff
@@ -1052,7 +1051,7 @@ def generate_step(params, batch):
     if not os.path.isdir(os.path.join(training_args.output_dir)):
         os.makedirs(os.path.join(training_args.output_dir), exist_ok=True)
 
-    def save_ckpt(ckpt_dir: str, commit_msg: str = ""):
+    def save_ckpt(ckpt_dir: str, commit_msg: str):
```
Comment (amyeroberts) [inline examples/flax/image-captioning/run_image_captioning_flax.py:1054]: We should keep the default here 

```suggestion
    def save_ckpt(ckpt_dir: str, commit_msg: str = ""):
```

**Review by amyeroberts — COMMENTED** [review]
Thanks for all the work on this and improving our examples! 

Just a few small changes to make, then we're good to merge!

File: examples/pytorch/token-classification/run_ner_no_trainer.py [hunk 2]
```diff
@@ -310,15 +310,8 @@ def main():
...
             # Create repo and retrieve repo_id
-            repo_id = create_repo(repo_name, exist_ok=True, token=args.hub_token).repo_id
-            # Clone repo locally
-            repo = Repository(args.output_dir, clone_from=repo_id, token=args.hub_token)
-
-            with open(os.path.join(args.output_dir, ".gitignore"), "w+") as gitignore:
-                if "step_*" not in gitignore:
-                    gitignore.write("step_*\n")
-                if "epoch_*" not in gitignore:
-                    gitignore.write("epoch_*\n")
```
[the remark this answers is not in the document]
Comment (Hvanderwilk) [reply examples/pytorch/token-classification/run_ner_no_trainer.py]: Alright, I've added it back and added it in an example which did not have it.

**Review by amyeroberts — APPROVED** [review]
Thanks for all this work improving our examples!

**Pull request merged**